lib/pulsar/broker.ex

```diff
@@ -699,23 +721,23 @@ defmodule Pulsar.Broker do
         :keep_state_and_data
 
       {consumer_pid, _monitor_ref} ->
-        Logger.info("Broker requested consumer #{consumer_id} closure")
+        Logger.warning("Broker requested consumer #{consumer_id} closure")
 
         send(consumer_pid, {:broker_message, command})
         :keep_state_and_data
     end
   end
 
-  defp handle_command(%Binary.CommandCloseProducer{producer_id: producer_id}, broker) do
+  defp handle_command(%Binary.CommandCloseProducer{producer_id: producer_id} = command, broker) do
     case Map.get(broker.producers, producer_id) do
       nil ->
         Logger.warning("Received close command for unknown producer #{producer_id}")
         :keep_state_and_data
 
       {producer_pid, _monitor_ref} ->
-        Logger.info("Broker requested producer #{producer_id} closure, will restart with fresh lookup")
+        Logger.info("Broker requested producer #{producer_id} closure")
 
-        Process.exit(producer_pid, :broker_close_requested)
+        send(producer_pid, {:broker_message, command})
```

Comment:
Makes sense, this allows us to keep some symmetry between producers [and consumers](https://github.com/efcasado/pulsar-elixir/blob/87220c9c9552d0f5149fd652a637314e0807ccc5/lib/pulsar/broker.ex#L710).